dolphin/alert_manager/trap_receiver.py

```diff
@@ -63,18 +64,63 @@ def _add_transport(self):
         except Exception:
             raise ValueError("Port binding failed the provided port is in use.")
 
+    def _extract_oid_value(self, var_bind):
+        """Extracts oid and value from var binds.
+        ex: varbind =  (SNMPv2-MIB::snmpTrapOID.0 = SNMPv2-MIB::authenticationFailure)
+        oid = snmpTrapOID
+        val = authenticationFailure
+        """
+
+        # Separate out oid and value strings
+        var_bind_info = var_bind.prettyPrint()
+        var_bind_info = var_bind_info.split("=", 1)
+        oid = var_bind_info[0]
+        val = var_bind_info[1]
+
+        # Extract oid from oid string
+        # Example: get snmpTrapOID from SNMPv2-MIB::snmpTrapOID.0)
+        oid = oid.split("::", 1)
```

Comment:
you could have simply used 're' module and written one line

Reply:
Updated